canine/adapters/base.py

```diff
@@ -69,22 +107,34 @@ def parse_inputs(self, inputs: typing.Dict[str, typing.Union[typing.Any, typing.
         Returns a job input specification useable for Localization
         Also sets self.spec to the same dictionary
         """
+
+        #Pin fixed arrays
+        inputs = {
+            key: _FixedArray(val) if key in self.arrays else val
+            for key,val in inputs.items()
+        }
+
         keys = sorted(inputs)
         input_lengths = {
-            key: len(val) if isinstance(val, list) else 1
+            # FixedArrays return actual length if they are 2d
+            key: len(val) if isinstance(val, list) or (isinstance(val, _FixedArray) and val.is_2d) else 1
```

Comment:
If a FixedArray cannot be created in the first place if it's not 2D, then we don't need these `val.is_2d` checks.

Reply:
A fixed array can be 1D

Reply:
Doh! I forgot about the possibility of FixedArrays being common inputs.